lib/stripe.rb

```diff
@@ -384,6 +384,12 @@ def self.handle_restclient_error(e, request_opts, retry_count, api_base_url=nil)
       message = "The connection to the server (#{api_base_url}) broke before the " \
         "request completed. #{connection_message}"
 
+    when OpenSSL::SSL::SSLError
+      message = "Could not establish a secure connection to Stripe, you may " \
+                "need to upgrade your OpenSSL version. To check, try running " \
+                "'openssl s_client -connect api.stripe.com:443' from the " \
+                "command line."
```

Comment:
This code actually falls through to https://github.com/stripe/stripe-ruby/pull/408/files#diff-7bd095922baff1048f819a71f5df2e2cL408, where we append the original error we received, so I think we're set :)